language-plutus-core/src/Language/PlutusCore/Universe/Default.hs

```diff
@@ -0,0 +1,94 @@
+-- | The universe used by default and its instances.
+
+{-# OPTIONS_GHC -fno-warn-orphans        #-}  -- The @Pretty ByteString@ instance.
+{-# OPTIONS_GHC -fno-warn-unused-matches #-}  -- Appears in generated instances.
+
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Language.PlutusCore.Universe.Default
+    ( DefaultUni (..)
+    ) where
+
+import           Language.PlutusCore.Pretty.Utils
+import           Language.PlutusCore.Universe.Core
+
+import qualified Data.ByteString.Lazy              as BSL
+import           Data.GADT.Compare.TH
+import           Data.Text.Prettyprint.Doc
+import           Language.Haskell.TH.Syntax
+
+-- | We want to pretty-print constants of built-in types in a particular way.
+-- Ideally, that should mean that either we have a particular class for constants pretty-printing
+-- or we use a newtype to wrap the types in, so that they can be assigned a fancy 'Pretty' instance.
+-- But for now we just hardcode an instance for 'ByteString'.
+instance Pretty BSL.ByteString where
+    pretty = prettyBytes
+
+{- Note [PLC types and universes]
+At the moment the default universe is finite and we don't have things like
+
+    DefaultUniList :: !(DefaultUni a) -> DefaultUni [a]
+
+Such a type constructor can be added, but note that this doesn't directly lead to interop between
+Plutus Core and Haskell, i.e. you can't have a meta-list whose elements are of a PLC type.
```

Comment:
This use of "meta-type" to refer to builtin types makes sense, but it makes me think that we need a note which lays out the general strategy for builtin types, which is to treat them as "meta-types" (i.e. host-language types).

Reply:
Extended `Note [PLC types and universes]` with

```
We encode built-in types in PLC as tags for Haskell types (the latter are also called meta-types),
see Note [Universes]. A built-in type in PLC is an inhabitant of

    Some (TypeIn uni)

where @uni@ is some universe, i.e. a collection of tags that have meta-types associated with them.

A value of a built-in type is a regular Haskell value stored in

    Some (ValueOf uni)

(together with the tag associated with its type) and such a value is also called a meta-constant.
```